src/filtering.ts

```diff
@@ -0,0 +1,71 @@
+import { logger } from "./logger"
+
+export type AIResponse = {
+  flagged: boolean
+  reason: string
+}
+
+export interface Message {
+  text: string
+  threadId: string
+  messageId: string
+  parent: string | null
+  flagged?: boolean
+  reason?: string
+}
+
+export interface UserMessage {
+  message: Message
+  timestamp: number
+  username: string
+  address: string
+}
+export async function callAI(prompt: string, userInput: string, url: string, token: string): Promise<AIResponse> {
+  const requestBody = {
+    messages: [
+      {
+        role: 'system',
+        content: prompt,
+      },
+      {
+        role: 'user',
+        content: `Comment to check: ${userInput}`,
+      },
+    ],
+  }
+
+  const controller = new AbortController()
+  const timeoutId = setTimeout(() => controller.abort(), 3000)
```

Comment:
is 3sec tested ? or just a starting value now and we'll see later ?

Reply:
Yes, 3 seconds tested, and it seems good. The timeout was tested with 3ms as well.

Reply:
@ferencsarai Sorry for not catching it earlier, but yes the [original requirements](https://app.zenhub.com/workspaces/spdevteam-6544d91246b817002d853e69/issues/zh/194) was to keep the timeout value as part of the config, so it can be easily changed as a parameter.

Reply:
@ferencsarai Can you add the timeout to the config? But if not quick fix, it can be in the next PR also.